app/src/main/java/de/danoeh/antennapod/dialog/FeedRefreshIntervalDialog.java

```diff
@@ -0,0 +1,161 @@
+                    case R.id.timeRadioButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.VISIBLE);
+
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            timePicker.setHour(finalHourOfDay);
+                        }
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                            timePicker.setMinute(finalMinute);
+                        }
+                        timePicker.setIs24HourView(DateFormat.is24HourFormat(context));
+                        break;
+                    case R.id.disableButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.GONE);
+                        break;
+                    default:
+
+                }
+            }
+        });
+
+        radioButtonInterval.setChecked(true);
+
+        builder.setPositiveButton(R.string.confirm_label, (dialog, which) -> {
+            if (radioButtonInterval.isChecked()) {
+                String text = spinner.getSelectedItem().toString();
+                String[] parts = text.split(" ");
+                int hours = Integer.parseInt(parts[0]);
+                UserPreferences.setUpdateInterval(hours);
+                onUpdateCallback.handleUpdate();
+                dialog.cancel();
+            } else if (radioButtonTime.isChecked()) {
+                int hour = 0;
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    hour = timePicker.getHour();
+                }
+                int minutx = 0;
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    minutx = timePicker.getMinute();
+                }
+
+                UserPreferences.setUpdateTimeOfDay(hour, minutx);
+                onUpdateCallback.handleUpdate();
+                dialog.cancel();
+            } else if (radioButtonDisable.isChecked()) {
+                UserPreferences.disableAutoUpdate(context);
+                onUpdateCallback.handleUpdate();
+            } else {
+                throw new IllegalStateException("Unexpected error.");
+            }
+        });
+
+        builder.setNegativeButton(R.string.cancel_label, null);
+        builder.show();
+    }
+
+    private static String[] appendHourUnit(Activity activity, final String[] values) {
+        final Resources res = activity.getResources();
+        String[] entries = new String[values.length];
+        for (int x = 0; x < values.length; x++) {
+            Integer v = Integer.parseInt(values[x]);
+            switch (v) {
+                case 0:
+                    entries[x] = res.getString(R.string.pref_update_interval_hours_manual);
+                    break;
+                case 1:
+                    entries[x] = v + " " + res.getString(R.string.pref_update_interval_hours_singular);
```

Comment:
Not sure if that works in all languages. How about using the plural string `time_hours_quantified` instead?